javascript/tables/src/validationUtils.ts

```diff
@@ -0,0 +1,324 @@
+		const valuesCasted =
+			dataType === DataType.String
+				? (values as string[])
+				: (values as Array<any>[])
+
+		if (includeInstances) {
+			valuesCasted.forEach((value: string | Array<any>, index: number) => {
+				if (!validate(value)) resultIndexes.push(index)
+			})
+			return {
+				fail: resultIndexes.length !== 0,
+				indexes: resultIndexes,
+				rule: ErrorCode.MaxLength,
+			}
+		} else {
+			return {
+				fail: !(dataType === DataType.String
+					? (values as string[]).every(validate)
+					: (values as Array<any>[]).every(validate)),
+				indexes: resultIndexes,
+				rule: ErrorCode.MaxLength,
+			}
+		}
+	}
+}
+
+export function validateMaxLength(
+	maxLength: number,
+): (value: Array<any> | string) => boolean {
+	return function validateMaxLength(value: Array<any> | string) {
+		return value.length <= maxLength
+	}
+}
+
+export function validateMinimumConstraint(
+	minimum: number,
+	dataType: DataType,
+): (values: unknown, includeInstances: boolean) => ValidationTestResult {
+	const validate = validateMinimum(minimum, dataType)
+	return function validateMinimumConstraint(
+		values: unknown,
+		includeInstances: boolean,
+	): ValidationTestResult {
+		const resultIndexes: number[] = []
+		const valuesCasted =
+			dataType === DataType.Number ? (values as number[]) : (values as Date[])
+
+		if (includeInstances) {
+			valuesCasted.forEach((value: number | Date, index: number) => {
+				if (!validate(value)) resultIndexes.push(index)
+			})
+			return {
+				fail: resultIndexes.length !== 0,
+				indexes: resultIndexes,
+				rule: ErrorCode.Minimum,
+			}
+		} else {
+			return {
+				fail: !(dataType === DataType.Number
+					? (values as number[]).every(validate)
+					: (values as Date[]).every(validate)),
+				indexes: resultIndexes,
+				rule: ErrorCode.Minimum,
+			}
+		}
+	}
+}
+
+export function validateMinimum(
+	minimum: number,
+	dataType: DataType,
+): (value: Date | number) => boolean {
+	return function validateMinimum(value: Date | number) {
+		return dataType == DataType.Date
+			? value.valueOf() >= minimum
+			: value >= minimum
+	}
+}
+
+export function validateMaximumConstraint(
+	maximum: number,
+	dataType: DataType,
+): (values: unknown, includeInstances: boolean) => ValidationTestResult {
+	const validate = validateMaximum(maximum, dataType)
+	return function validateMaximumConstraint(
+		values: unknown,
+		includeInstances: boolean,
+	): ValidationTestResult {
+		const resultIndexes: number[] = []
+		const valuesCasted =
+			dataType === DataType.Number ? (values as number[]) : (values as Date[])
+
+		if (includeInstances) {
+			valuesCasted.forEach((value: number | Date, index: number) => {
+				if (!validate(value)) resultIndexes.push(index)
+			})
+			return {
+				fail: resultIndexes.length !== 0,
+				indexes: resultIndexes,
+				rule: ErrorCode.Maximum,
+			}
+		} else {
+			return {
+				fail: !(dataType === DataType.Number
+					? (values as number[]).every(validate)
```

Comment:
Can you cast this to `(values as number[] || Date[])` to remove the logical switch?

Reply:
I tried it but didn't work